feature/experimental/p4rt/otg_tests/traceroute_packetin_test/README.md

```diff
@@ -1,34 +1,243 @@
 
-## Config Parameter coverage
+    ```
+    IPv4Entry {192.51.100.1/24 (DECAP_TE_VRF)} -> NHG#1001 (DEFAULT VRF) -> {
+        {NH#1001, DEFAULT VRF, weight:1}
+    }
+    NH#1001 -> {
+        decapsulate_header: OPENCONFIGAFTTYPESDECAPSULATIONHEADERTYPE_IPV4
+    }
 
-*    /components/component/integrated-circuit/config/node-id
-*    /interfaces/interface/config/id
+    ```
 
+2.  Apply vrf selection policy `vrf_selection_policy_w` to DUT port-1.
 
-## Telemetry Parameter coverage
+3.  Send the following 6in4 and 4in4 flows to DUT port-1:
 
-No new telemetry covered.
+    ```
+    * inner_src: `ipv4_inner_src`
+    * inner_dst: `ipv4_inner_encap_match`
+    * dscp: `dscp_encap_no_match`
+    * outer_src: `ipv4_outer_src_111`
+    * outer_dst: `ipv4_outer_decap_match`
+    * dscp: `dscp_encap_no_match`
+    * proto: `4`
+	* outer TTL: `1`
 
+    * inner_src: `ipv6_inner_src`
+    * inner_dst: `ipv6_inner_encap_match`
+    * dscp: `dscp_encap_no_match`
+    * outer_src: `ipv4_outer_src_111`
+    * outer_dst: `ipv4_outer_decap_match`
+    * dscp: `dscp_encap_no_match`
+    * proto: `41`
+	* outer TTL: `1`
+    ```
 
-## Protocol/RPC Parameter coverage
+4.  Verify that all punted packets:
+    *   Have ingress_port and target_egress_port metadata set
+    *   target_egress_port is set to DUT port-8 per the hierarchical weight.
 
-No new Protocol/RPC covered.
+6.  Change the subnet mask from /24 and repeat the test for the masks  /32, /22, and /28 and verify again that the packets are punted correctly.
 
+
+### TODO: Test-7 (TBD)
+
+Encap failure cases (TBD on confirmation)
+
+### TODO: Test-8 (TBD)
+
+Tests that traceroute for a packet with a route lookup miss has an unset target_egress_port.
+
+### TODO: Test-9, decap the encap
+
+1.  Using gRIBI to install the following entries in the `DECAP_TE_VRF`:
+
+    ```
+    IPv4Entry {192.51.100.1/24 (DECAP_TE_VRF)} -> NHG#1001 (DEFAULT VRF) -> {
+        {NH#1001, DEFAULT VRF, weight:1}
+    }
+    NH#1001 -> {
+        decapsulate_header: OPENCONFIGAFTTYPESDECAPSULATIONHEADERTYPE_IPV4
+    }
+    ```
+
+2.  Apply vrf selection policy `vrf_selection_policy_w` to DUT port-1.
+
+3.  Send the following packets to DUT port-1:
+
+    ```
+    * inner_src: `ipv4_inner_src`
+    * inner_dst: `ipv4_inner_encap_match`
+    * dscp: `dscp_encap_a_1`
+    * outer_src: `ipv4_outer_src_222`
+    * outer_dst: `ipv4_outer_decap_match`
+    * dscp: `dscp_encap_a_1`
+    * proto: `4`
+    * outer TTL: '1'
+    ```
+
+    ```
+    * inner_src: `ipv6_inner_src`
+    * inner_dst: `ipv6_inner_encap_match`
+    * dscp: `dscp_encap_a_1`
+    * outer_src: `ipv4_outer_src_111`
+    * outer_dst: `ipv4_outer_decap_match`
+    * dscp: `dscp_encap_a_1`
+    * proto: `41`
+	* outer TTL: `1`
+    ```
+
+4.  We should expect that all punted packets:
+    *   Have ingress_port and target_egress_port metadata set
+    *   target_egress_port is set to DUT port-2, port-3, port-4 and port-6 per the hierarchical weight.
```

Comment:
Please check this also it will choose port 8 to out as DECAP_TE_VRF point NH to Default vrf